internal/cache/file/downloader/job.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package downloader
+
+import (
+	"container/list"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/cache/lru"
+	"github.com/googlecloudplatform/gcsfuse/internal/locker"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/gcs"
+	"golang.org/x/net/context"
+)
+
+type jobStatusName string
+
+const (
+	NOT_STARTED jobStatusName = "NOT_STARTED"
+	DOWNLOADING jobStatusName = "DOWNLOADING"
+	COMPLETED   jobStatusName = "COMPLETED"
+	FAILED      jobStatusName = "FAILED"
+	CANCELLED   jobStatusName = "CANCELLED"
+)
+
+// Job downloads the requested object from GCS into the specified local file
+// path with given permissions and ownership.
+type Job struct {
+	/////////////////////////
+	// Constant data
+	/////////////////////////
+
+	object               *gcs.MinObject
+	bucket               gcs.Bucket
+	filePath             string
+	fileInfoCache        *lru.Cache
+	sequentialReadSizeMb int32
+	filePerm             os.FileMode
```

Comment:
Should we club all file_creation parameter as a struct? E.g. LocalFileSpec { filePath, filePerm, fileUid, fileGid}